piAIOComm.c

```diff
@@ -268,23 +268,23 @@ u32 piAIOComm_Init(u8 devnum)
 	pr_info_aio("piAIOComm_Init send configIn1\n");
 	ret = pibridge_req_io(addr, IOP_TYP1_CMD_DATA2, snd_buf,
 			      AIO_CONFIG_DATA2_LEN, NULL, 0);
-	if (ret)
+	if (ret < 0)
```

Comment:
I think you don't need to change anything here. You're expecting an empty response (payload with 0 bytes). So receiving anything greater than 0 would be an error.